### PR TITLE
Delete duplicated content, add redirects, minor clean up

### LINKED ISSUE #19932: Consolidate "State and data in Docker applications"
https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/architect-microservice-container-applications/docker-application-state-data.md
https://github.com/dotnet/docs/blob/master/docs/architecture/containerized-lifecycle/design-develop-containerized-apps/state-and-data-in-docker-applications.md

These two articles seems to be nearly identical. I think they need to be consolidated into one.

Related to https://github.com/dotnet/docs/pull/19929

### LINKED ISSUE #19933: Consolidate "Docker containers, images, and registries"
https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/container-docker-introduction/docker-containers-images-registries.md

https://github.com/dotnet/docs/blob/master/docs/architecture/containerized-lifecycle/docker-containers-images-and-registries.md

Related to #19929

### LINKED ISSUE #19934: Consolidate "Docker terminology"
https://github.com/dotnet/docs/blob/master/docs/architecture/containerized-lifecycle/docker-terminology.md

https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/container-docker-introduction/docker-terminology.md

Related to #19929

### LINKED ISSUE #19935: Consolidate "What is Docker"
https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/container-docker-introduction/docker-defined.md

https://github.com/dotnet/docs/blob/master/docs/architecture/containerized-lifecycle/what-is-docker.md

Related to #19929

### LINKED ISSUE #25380: Duplicate content in architecture e-books
This issue is to track getting to a clean build in the architecture folder. Currently there are warnings for duplicated H1 headings and metadata titles in ~10 articles in this folder. To resolve the warnings, we could either differentiate the H1s and titles, or we could consolidate duplicated content (as suggested by #19932, #19933, #19934, #19935).

Related to #25322.

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Delete duplicated content, add redirects, minor clean up

All this:
- Fixes #19932
- Fixes #19933
- Fixes #19934
- Fixes #19935
- Fixes #25380
- Related to #25322

### CODE REVIEW

File: docs/architecture/containerized-lifecycle/toc.yml [hunk 1]
```diff
@@ -2,18 +2,16 @@
...
     - name: What is Docker?
-      href: what-is-docker.md
+      href: ../microservices/container-docker-introduction/docker-defined.md?toc=/dotnet/architecture/containerized-lifecycle/toc.json&bc=/dotnet/breadcrumb/toc.json
     - name: Docker terminology
-      href: docker-terminology.md
+      href: ../microservices/container-docker-introduction/docker-terminology.md?toc=/dotnet/architecture/containerized-lifecycle/toc.json&bc=/dotnet/breadcrumb/toc.json
     - name: Docker containers, images, and registries
-      href: docker-containers-images-and-registries.md
-    - name: Road to modern applications based on containers
-      href: road-to-modern-applications-based-on-containers.md
```
Comment (gewarren) [inline docs/architecture/containerized-lifecycle/toc.yml:14]: Did you mean to remove this entry completely?

**Review by gewarren — COMMENTED** [review]
Looking good. I think you'll still need to add dual Previous/Next buttons though, since they'll be different for different ebooks.

File: docs/architecture/containerized-lifecycle/toc.yml (hunk 1 above)
Comment (IEvangelist) [reply docs/architecture/containerized-lifecycle/toc.yml:14]: Yes, the content was rather small and it didn't seem to add value. It had two links to both e-books that contain the shared content - and brief description of each.

**Comment by IEvangelist** [comment]
Hi @gewarren,

In a conversation with @nishanil - we collectively decided based on a few internal limitations, concerning the tooling that generates the e-book PDFs that it's best to allow duplication in these few situations. We are all aware of the number of articles, and the potential that they may drift apart. We're ok with that, knowing that they are separate e-books who are simply both relying on similar starting content, but contextually different. Drift is not only ok, it's actually encouraged. I'm going to close this draft, and all the related issues.

**Pull request closed without merging**